redfish-core/lib/power_supply_metrics.hpp

```diff
@@ -0,0 +1,370 @@
+ * After getting maximum values, proceed to populating Redfish JSON response
+ * properties.
+ *
+ * @param[in/out] aResp - Shared pointer for asynchronous calls.
+ * @param[in] serviceName - The service providing the Maximum interface.
+ * @param[in] maximumPath - The object path the Maximum interface is on.
+ * @param[in] averageValues - Populated vector of date/time and average values.
+ *
+ * @return None.
+ */
+inline void getMaximumValues(const std::shared_ptr<bmcweb::AsyncResp>& aResp,
+                             const std::string& serviceName,
+                             const std::string& maximumPath,
+                             const averageMaxArray& averageValues)
+{
+    // gets values from Maximum interface, call parseAverageMaximum() to plop
+    // in the array.
+    // Do code or call code to push up into JSON response.
+
+    BMCWEB_LOG_DEBUG << "Get Values from serviceName: " << serviceName
+                     << " objectPath: " << maximumPath
+                     << " interfaceName: " << maximumInterface;
+
+    crow::connections::systemBus->async_method_call(
+        [aResp, serviceName, maximumPath, averageValues](
+            const boost::system::error_code ec2,
+            const std::variant<averageMaxArray>& intfValues) mutable {
+            if (ec2)
+            {
+                BMCWEB_LOG_DEBUG << "D-Bus response error";
+                messages::internalError(aResp->res);
+                return;
+            }
+            const averageMaxArray* intfValuesPtr =
+                std::get_if<averageMaxArray>(&intfValues);
+            if (intfValuesPtr == nullptr)
+            {
+                messages::internalError(aResp->res);
+                return;
+            }
+
+            averageMaxArray maximumValues;
+
+            for (const auto& values : *intfValuesPtr)
+            {
+                // The first value returned is the timestamp, it is in
+                // milliseconds since the Epoch.
+                auto dateTime = std::get<0>(values);
+                // The second value returned is in watts. The maximum watts this
+                // power supply has used in a 30 second interval.
+                auto value = std::get<1>(values);
+
+                BMCWEB_LOG_DEBUG
+                    << "Date/Time: "
+                    << crow::utility::getDateTime(
+                           static_cast<std::time_t>(dateTime / 1000));
+                BMCWEB_LOG_DEBUG << "Maximum Value: " << value;
+
+                maximumValues.emplace_back(dateTime, value);
+            }
+
+            parseAverageMaximum(aResp, averageValues, maximumValues);
+        },
+        serviceName, maximumPath, "org.freedesktop.DBus.Properties", "Get",
+        maximumInterface, "Values");
+}
+
+/**
+ * @brief Gets the values from the Average interface and populates array.
+ *
+ * After getting average values, proceed to get maximum values.
+ *
+ * @param[in] aResp - Shared pointer for asynchronous calls.
```

Comment:
Missing doxygen for the other 3 parameters.

Reply:
Ack